test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2024 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-datetime-format-functions
+description: >
+  Temporal objects cannot be formatted if there is no overlap between the
+  provided options and the data model of the object
+features: [Temporal]
+---*/
+
+const sampleEpochMs = 1726773817847;  /* 2024-09-19T19:23:37.847Z */
+const sampleEpochNs = BigInt(sampleEpochMs) * 1_000_000n;
+const legacyDate = new Date(sampleEpochMs);
+const instant = new Temporal.Instant(sampleEpochNs);
+const plainDateTime = new Temporal.PlainDateTime(2024, 9, 19, 12, 23, 37, 847);
+const plainDate = new Temporal.PlainDate(2024, 9, 19);
+const plainYearMonth = new Temporal.PlainYearMonth(2024, 9);
+const plainMonthDay = new Temporal.PlainMonthDay(9, 19);
+const plainTime = new Temporal.PlainTime(12, 23, 37, 847);
+
+const dateStyleFormatter = new Intl.DateTimeFormat(undefined, { dateStyle: "short", calendar: "iso8601", timeZone: "America/Vancouver" });
+const dateStyleResult = JSON.stringify(dateStyleFormatter.formatRangeToParts(legacyDate, legacyDate));
```

Comment:
Is the serialization order guaranteed (or reliable enough?)

Reply:
It's "ascending chronological order of property creation" and the order in which `type`, `value`, and `source` are created is specified, so I think we're good.